### PR TITLE
Fix PAGE_SIZE changes from #10284 

### PR DESCRIPTION
**Pull request by kripken**
The wasm/asmjs page size is different from the filesystem page size, I think we mixed them up a little before.

This should also be smaller than before as it avoids calling `getpagesize()`.

Alternative to #10318

### CODE REVIEW

File: src/library.js [hunk 1]
```diff
@@ -300,34 +300,20 @@ LibraryManager.library = {
...
-#if MINIMAL_RUNTIME
-#if WASM
-    return {{{ WASM_PAGE_SIZE }}};
-#else
-    return {{{ ASMJS_PAGE_SIZE }}};
-#endif
-#else
-    return PAGE_SIZE;
-#endif
+    return {{{ FS_PAGE_SIZE }}};
```
Comment (sbc100) [inline src/library.js]: getpagesize is a posix function that is supposed to return virtual memory page size.. so why is returning the VFS one?

File: src/parseTools.js [hunk 2]
```diff
@@ -1546,6 +1546,9 @@ function makeRetainedCompilerSettings() {
   return ret;
 }
 
+// Page size of the virtual filesystem.
+var FS_PAGE_SIZE = 16384;
```
Comment (sbc100) [inline src/parseTools.js]: Seems like it s not just the filesystem but also mmap that uses this.

How about calling this `USER_PAGE_SIZE`?   Or mabye something more meaningfull.  I'm not sure that the purpose of it is exactly.

File: src/library.js (hunk 1 above)
Comment (sbc100) [reply src/library.js]: For example won't brk()/malloc() use this value?

Comment (kripken) [reply src/library.js]: This restores the behavior before that PR, so this is just a revert really. (But it also refactors the value into something we can `{{{ }}}` for efficiency.)

This page size is used in the filesystem layer, yeah. I don't actually think it matters much in practice - malloc won't use it, as we use sbrk() there, and it doesn't limit itself to pages. But perhaps user code might, so I went back to the behavior we've had all along.

I don't remember well why this value was picked all the years ago, but I assume it was to make it similar to page sizes in existing OSes?

File: src/parseTools.js (hunk 2 above)
Comment (kripken) [reply src/parseTools.js]: Happy to call it `USER_PAGE_SIZE`, but it is used in the filesystem layer and nowhere else, so I think `FS_` makes sense? (the old name for it was just `PAGE_SIZE` which was ambiguous)

Comment (sbc100) [reply src/parseTools.js]: I looks like it used by `getpagesize` and by `mmap` .. I guess you could argue that mmap is FS related but not solely (MAP_ANON etc).

Comment (kripken) [reply src/parseTools.js]: Hmm, yeah. We don't support MAP_ANON though, so it's purely FS for now.

I think FS is better than USER, as USER is more ambiguous. How strongly do you feel about this?

Comment (sbc100) [reply src/parseTools.js]: I don't feel super strongly.   getpagesize is nothing to do with filesystem though.

How about REPORTED_PAGE_SIZE?  Or POSIX_PAGE_SIZE?

Comment (sbc100) [reply src/parseTools.js]: lgtm either way

Comment (kripken) [reply src/parseTools.js]: I like `POSIX_PAGE_SIZE`, switched to that. Added a clarifying comment too.

**Pull request merged**